ab3d2_source/c/system.c

```diff
@@ -0,0 +1,421 @@
+#include "system.h"
+#include "screen.h"
+#include "draw.h"
+
+#include <SDI_compiler.h>
+#include <SDI_misc.h>
+#include <cybergraphics/cybergraphics.h>
+#include <devices/timer.h>
+#include <dos/dos.h>
+#include <exec/exec.h>
+#include <graphics/gfxbase.h>
+#include <hardware/intbits.h>
+#include <proto/cybergraphics.h>
+#include <proto/dos.h>
+#include <proto/exec.h>
+#include <proto/graphics.h>
+#include <proto/intuition.h>
+#include <proto/lowlevel.h>
+#include <proto/misc.h>
+#include <proto/potgo.h>
+#include <proto/timer.h>
+#include <resources/misc.h>
+#include <resources/potgo.h>
+
+#define ALIB_HARDWARE_CUSTOM
+#include <clib/alib_protos.h>
+#include <proto/alib.h>
+
+#define INTUITIONNAME "intuition.library"
+
+extern struct TimeRequest sys_TimerRequest;
+extern struct EClockVal Sys_FrameTimeECV_q[2];
+extern ULONG Sys_ECVToMsFactor_l;
+extern ULONG Sys_FrameTimes_vl[8];
+extern ULONG Sys_FrameNumber_l;
+extern UWORD Sys_FPSIntAvg_w;
+extern UWORD Sys_FPSFracAvg_w;
+
+extern UBYTE Sys_Move16_b;
+extern UBYTE Vid_FullScreen_b;
+extern struct EClockVal _Sys_FrameTimeECV_q[2];
+
+extern UWORD angpos;
+extern WORD Sys_MouseY;
+
+extern UBYTE KeyMap_vb[256];
+
+static BOOL gotSerialPort;
+static BOOL gotSerialBits;
+static UWORD allocPotBits;
+
+struct Library *CYBERGFX_BASE_NAME = NULL;
+
+extern VOID VBlankInterrupt(void);
+extern VOID key_interrupt(void);
+
+static const char AppName[] = "TheKillingGrounds";
+static struct Interrupt VBLANKInt = {{NULL, NULL, NT_INTERRUPT, 9, (char *)AppName}, 0, VBlankInterrupt};
+static struct Interrupt KBInt = {NULL, NULL, NT_INTERRUPT, 127, (char *)AppName, 0, key_interrupt};
+
+static struct MsgPort vblPort;
+static struct VBLData
+{
+    volatile ULONG tsi_Flag;
+    struct MsgPort *tsi_Port;
+} vblData = {0, &vblPort};
+
+static BOOL SAVEDS FakeVBlankInterrupt(REG(a1, struct VBLData *data));
+
+static struct Interrupt PseudoVBLANKInt = {
+    NULL, NULL, NT_INTERRUPT, 0, (char *)AppName, &vblData, &FakeVBlankInterrupt};
+
+static BOOL sys_InitHardware();
+static void sys_ReleaseHardware();
+static void sys_InstallInterrupts();
+static void sys_RemoveInterrupts();
+
+BOOL Sys_Init()
+{
+    if (!sys_InitHardware()) {
+        goto fail;
+    }
+
+    ULONG freq = Sys_MarkTime(&Sys_FrameTimeECV_q[1]);
+    Sys_ECVToMsFactor_l = (1000 << 16) / freq;
+
+    sys_InstallInterrupts();
+
+    if (!Draw_Init()) {
+        goto fail;
+    }
+
+    return TRUE;
+
+fail:
+    sys_ReleaseHardware();
+    return FALSE;
+}
+
+void Sys_Done()
+{
+    Draw_Shutdown();
+    sys_RemoveInterrupts();
+    sys_ReleaseHardware();
+}
+
+BOOL sys_OpenLibs(void)
```

Comment:
Mentioned elsewhere, but we should capitalise this now that it's called from the "outside"...